src/python/pants/source/source_root.py

```diff
@@ -98,6 +102,12 @@ def find_by_path(self, path: str) -> Optional[SourceRoot]:
     # TODO: Remove this logic. It should be an error to have no matching source root.
     return SourceRoot(path, (), SourceRootCategories.UNKNOWN)
 
+  # TODO: this is how find_by_path should behave. Figure out how to deprecate the behavior of
+  # find_by_path so that this method becomes redundant.
```

Comment:
`find_by_path` will return the original path when there is no source root for that path. 

Given `no-source-root/example.txt`, for example, this meant that we tried stripping `no-source-root/example.txt` from `no-source-root/example.txt`.